src/cpp/include/aiebu/aiebu_assembler.h

```diff
@@ -152,53 +164,81 @@ class aiebu_assembler
+     * ELF as well.
+     *
      * When using this class to do xrt argument transform
      *      1. Only host patching is supported.
      *      2. Only support AIE2PS and AIE4
      */
-    class argtbl_impl;
-    class argtbl
-    {
-      private:
-        std::shared_ptr<argtbl_impl> handle;
-      public:
-        explicit argtbl(std::shared_ptr<argtbl_impl> in_impl);
-
-        /*
-         * Dump the reference of vetor of xrt argument (xrt_idx)
-         * and BD offset. Caller can modify entries in the vector
-         * in place. Then the whole argtbl object can be flushed
-         * back to aiebu_assembler to update the ELF to do the
-         * xrt_idx transform.
-         */
-        std::vector<arginfo>& dump() const;
-
-    };
-    /*
-     * Get an argtbl object from aiebu_assembler. In this function
-     * aiebu_assember will scan the ELF and construct a table (vector)
-     * of xrt_idx and BD offset. Then it returns an object of this table
-     * which can be used to dump the reference of the table and modify
-     * the table in place.
-     */
-    argtbl get_argtbl();
-
-    /*
-     * Flush the argtbl object to aiebu_assembler. In this function,
-     * aiebu will take the argtbl object and update control code and
-     * patching metadata based on the xrt_idx and BD offset in the
-     * arg table.
-     */
-    void flush_argtbl(const argtbl& arg_table);
+     class argtbl
+     {
+         friend class aiebu_assembler;
+       private:
+         std::shared_ptr<argtbl_impl> handle;
+       public:
+         explicit argtbl(std::shared_ptr<argtbl_impl> in_impl);
+
+         /*
+          * Get the reference of vector of instance info. Inside
+          * each instance, there is table of xrt argument (xrt_idx)
+          * and BD offset of that instance. Caller can modify entries
+          * in the table in place. Then the whole argtbl object can be flushed
+          * back to aiebu_assembler to update the ELF to do the
+          * xrt_idx transform.
+          */
+         std::vector<instinfo>& get();
+
+         /*
+          * Update the kernel name in the instance info object
+          * @param name: the new kernel name (e.g., "NewKernel")
+          *
+          * The kernel name in ELF won't be updated until the whole
+          * object is flushed back.
+          */
+         void set_name(const std::string& name);
+     };
+
+     /*
+      * Get an argtbl object from aiebu_assembler for a given
+      * kernel name. In this function aiebu_assember will scan
+      * the ELF and construct a vector of instance info (instinfo).
+      * Inside each instinfo, we have a table (vector)
+      * of xrt_idx and BD offset of that kernel:inst's control code.
+      *
+      * The object can be used to dump the reference of the instance
+      * info and modify the table in place inside each instance.
+      *
+      * NOTE: applicable for only config elf's
```

Comment:
We don't really have a official concept of "config elf's". Are you trying to say it is only available for full ELF flow?